### ISSUE TITLE
[Point Cloud] Processing speed (DGCNN - ModelNet40)

### ISSUE THREAD

**Issue by czha5168** [rusty1s/pytorch_geometric#694]
## ❓ Questions & Help

Dear Matthias, 
Thanks for your contribution to the entire community. This is a really significant project/repo!
I am currently working on the point cloud study and am willing to adopt your repo for the further step. During the process, I noticed a relatively low processing speed (e.g., perform classification on modelnet40 dataset using dgcnn), compared to the [version released by the author, Yue](https://github.com/WangYueFt/dgcnn). Now, I am trying to reduce this difference and was wondering could you please provide me with some comments to guide me on it, as the designer and developer of this beautiful repo? Your comments are much appreciated!

My guess/attempts:
_Data loading:_ They use hdf5, rather than pt file to store the point cloud data. (BTW, they use same 2048 points sampled in previous pointnet research work, to make a fair comparison).
_Model Construction:_ I have updated the dgcnn to match their architecture claimed in the paper, based on your useful example (dgcnn_classification). And now it shares the same #params as Yue's implementation (1.81M), but the training time was correspondingly much longer (almost 8x compared to Yue's work).
_Others:_ Transformation(random scale, translation, etc) and other low-level modules (like DataParallel) might be improved?

Thanks in advance and looking forward to your suggestions. Have a good day!

**Comment by czha5168** [comment]
Hi Matthias, just a quick update that I think the dataloading & transform might not be the issue, as the DataListLoader can be created in roughly 0.05s. Thanks.

**Comment by rusty1s** [comment]
Hi, Thank you for looking over this. I would very much appreciate any pull request once the current version has been improved. I am quite sure that the main advantage of the official implementation is the efficient knn-processing. Instead of computing an edge tensor of shape [num_nodes, k, num_features], we use the MessagePassing interface to aggregate localized information. This interface works for any number of neighbors, but might be hence not the best choice when the number of neighbors is fixed for each node. So, replacing this operator should result in a huge runtime improvement. I can look into this if you like.

**Comment by czha5168** [comment]
> Hi, Thank you for looking over this. I would very much appreciate any pull request once the current version has been improved. I am quite sure that the main advantage of the official implementation is the efficient knn-processing. Instead of computing an edge tensor of shape [num_nodes, k, num_features], we use the MessagePassing interface to aggregate localized information. This interface works for any number of neighbors, but might be hence not the best choice when the number of neighbors is fixed for each node. So, replacing this operator should result in a huge runtime improvement. I can look into this if you like.

I see. Many thanks for replying me. I will keep you updated if any progress is made later.
